File: backend/moonstreamapi/reporter.py
import uuid

from humbug.consent import HumbugConsent
from humbug.report import HumbugReporter

from .settings import HUMBUG_REPORTER_BACKEND_TOKEN

session_id = str(uuid.uuid4())
client_id = "moonstream-backend"

reporter = HumbugReporter(
    name="moonstreamapi",
    consent=HumbugConsent(True),
    client_id=client_id,
    session_id=session_id,
    bugout_token=HUMBUG_REPORTER_BACKEND_TOKEN,
    tags=[],
)

File: crawlers/ldb/cmd/reporter.go
package cmd

import (
	"encoding/json"
	"fmt"

	"github.com/bugout-dev/humbug/go/pkg"
	"github.com/bugout-dev/moonstream/crawlers/ldb/configs"
)

var (
	humbugReporter *HumbugReporter
)

// Generate humbug client
func setHumbugClient(sessionID string) error {
	consent := humbug.CreateHumbugConsent(humbug.True)
	reporter, err := humbug.CreateHumbugReporter(consent, configs.HUMBUG_LDB_CLIENT_ID, sessionID, configs.HUMBUG_LDB_TOKEN)
	if err != nil {
		return fmt.Errorf("Unable to generate humbug reporter: %v", err)
	}
	humbugReporter.Reporter = reporter

	return nil
}

// Publish report with verified blocks to entry
func (r *HumbugReporter) submitReport(start, end uint64, prefix string) error {
	content, err := json.Marshal(corruptBlocks)
	if err != nil {
		return fmt.Errorf("Unable to marshal to json: %v", err)
	}

	report := humbug.Report{
		Title:   fmt.Sprintf("%sLDB verifier %d-%d", prefix, start, end),
		Content: string(content),
		Tags: []string{
			fmt.Sprintf("start:%d", start),
			fmt.Sprintf("end:%d", end),
		},
	}
	r.Reporter.Publish(report)
	fmt.Printf("%sLDB report for range %d-%d published\n", prefix, start, end)

	return nil
}
